src/java/org/apache/cassandra/tools/nodetool/SetAutoRepairConfig.java

```diff
@@ -0,0 +1,156 @@
+import io.airlift.airline.Arguments;
+import io.airlift.airline.Command;
+import io.airlift.airline.Option;
+import org.apache.cassandra.locator.InetAddressAndPort;
+import org.apache.cassandra.repair.autorepair.AutoRepairConfig.RepairType;
+import org.apache.cassandra.tools.NodeProbe;
+import org.apache.cassandra.tools.NodeTool.NodeToolCmd;
+
+import java.io.PrintStream;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+@Command(name = "setautorepairconfig", description = "sets the autorepair configuration")
+public class SetAutoRepairConfig extends NodeToolCmd
+{
+    @VisibleForTesting
+    @Arguments(title = "<autorepairparam> <value>", usage = "<autorepairparam> <value>",
+    description = "autorepair param and value.\nPossible autorepair parameters are as following: " +
+                  "[threads|subranges|minrepairintervalinhours|sstablehigherthreshold" +
+                  "|enabled|tablemaxrepairtimeinsec|priorityhost|forcerepairhosts|ignoredcs" +
+                  "|historydeletehostsclearbufferinsec|primarytokenrangeonly|parallelrepaircount|parallelrepairpercentage|mvrepairenabled]",
+    required = true)
+    protected List<String> args = new ArrayList<>();
+
+    @VisibleForTesting
+    @Option(title = "repair type", name = { "-t", "--repair-type" }, description = "Repair type")
+    protected RepairType repairType;
+
+    @VisibleForTesting
+    protected PrintStream out = System.out;
+
+    @Override
+    public void execute(NodeProbe probe)
+    {
+        checkArgument(repairType != null, "--repair-type is required.");
+        checkArgument(args.size() == 2, "setautorepairconfig requires param-type, and value args.");
+        String paramType = args.get(0);
+        String paramVal = args.get(1);
+
+        if (!probe.getAutoRepairConfig().isAutoRepairSchedulingEnabled())
+        {
+            out.println("Auto-repair is not enabled");
+            return;
+        }
+
+        if (paramType.equals("historydeletehostsclearbufferinsec"))
+        {
+            probe.setAutoRepairHistoryClearDeleteHostsBufferInSec(Integer.parseInt(paramVal));
+            return;
+        }
+
+        // options below require --repair-type option
+        checkArgument(repairType != null, "--repair-type is required for this parameter.");
+        Set<InetAddressAndPort> hosts;
+        switch (paramType)
+        {
+            case "enabled":
+                probe.setAutoRepairEnabled(repairType, Boolean.parseBoolean(paramVal));
+                break;
+            case "threads":
+                probe.setRepairThreads(repairType, Integer.parseInt(paramVal));
+                break;
+            case "subranges":
+                probe.setRepairSubRangeNum(repairType, Integer.parseInt(paramVal));
+                break;
+            case "minrepairintervalinhours":
```

Comment:
One thing I just stumbled into is that the formatting of multi-word config here is different than the yaml configuration, e.g. `minrepairinternvalinhours` in the yaml is `min_repair_internval_in_hours`.  Can we update the nodetool config to match the yaml one?

Reply:
Sure, will do

Reply:
Done